src/App.php

```diff
@@ -144,12 +148,45 @@ class App
     /** @var class-string */
     public $templateClass = HtmlTemplate::class;
 
+    /** @var ResponseInterface */
+    private $response;
+
+    /** @var ServerRequestInterface */
+    private $request;
+
     public function __construct(array $defaults = [])
     {
         $this->setApp($this);
 
+        $this->response = $defaults['response'] ?? new Response();
+        if (isset($defaults['response'])) {
+            unset($defaults['response']);
+        }
+
+        // TODO remove this before PSR7 PR merge, fix tests to pass /wo this develop merge hack
```

Comment:
TODO before merge

Reply:
@abbadon1334 I have merged develop branch in and removed this TODO, please adjust the code accordingly to make CI gree again